lib/game.rb

```diff
@@ -0,0 +1,92 @@
+
+
+class Game
+  WIN_COMBINATIONS = [[0,1,2],[3,4,5],[6,7,8],[0,3,6],[1,4,7],[2,5,8],[0,4,8],[2,4,6]]
+
+  attr_accessor :board, :player_1, :player_2
+
+  def initialize(player_1 = Players::Human.new("X"), player_2 = Players::Human.new("O"), board = Board.new)
+    @board = board
+    @player_1 = player_1
+    @player_2 = player_2
+  end
+
+  def current_player
+    @board.turn_count.even? ? @player_1 : @player_2
+  end
+
+  def over?
+    draw? || won?
+  end
+
+  def won?
+    answer = []
+    WIN_COMBINATIONS.each do |combo|
+      result = []
+      combo.each { |j| result << @board.cells[j] }
+      answer << result
+      break if result.all?{ |ele| ele == "X" } || result.all?{ |ele|  ele == "O"}
+      end
+      answer.last.all?{ |ele| ele == "X" } || answer.last.all?{ |ele|  ele == "O"}
+  end
+
+  def draw?
+    !won? && @board.full?
+  end
+
+  def winner
+    if won?
```

Comment:
Consider using the following for this method:

```ruby
if won = won?
  board.cells[won.first]
end
```

If `won?` returns true, then and only then will the local variable `won` be assigned, otherwise, the method will just abort with a false return value. But if `won` is assigned, we can use that variable instead of calling `won?` again within the if statement. You won't have to create separate methods to check the winner.